platform/publisher.go

```diff
@@ -23,46 +23,47 @@
 package platform
 
 import (
-	"crypto/md5"
-	"encoding/hex"
+	"os"
+
 	"github.com/JetBrains/qodana-cli/v2025/cloud"
 	"github.com/JetBrains/qodana-cli/v2025/platform/qdenv"
 	"github.com/JetBrains/qodana-cli/v2025/platform/utils"
+	"github.com/JetBrains/qodana-cli/v2025/tooling"
 	log "github.com/sirupsen/logrus"
-	"io"
-	"net/http"
-	"os"
-	"path/filepath"
 )
 
-const PublisherJarName = "publisher-cli.jar"
-const PublisherVersion = "2.1.31"
-
 type Publisher struct {
 	ResultsDir string
 	LogDir     string
 	AnalysisId string
 }
 
 // SendReport sends report to Qodana Cloud.
-func SendReport(publisher Publisher, token string, publisherPath string, javaPath string) {
-	if _, err := os.Stat(publisherPath); os.IsNotExist(err) {
-		err := os.MkdirAll(filepath.Dir(publisherPath), os.ModePerm)
-		if err != nil {
-			log.Fatalf("failed to create directory: %v", err)
-		}
-		fetchPublisher(publisherPath)
+func SendReport(publisher Publisher, token string, javaPath string) {
+	fp, err := os.CreateTemp("", "qodana-publisher")
```

Comment:
qodana-publisher.jar?

Reply:
Fixed, though note that CreateTemp appends a random suffix to the file, so it would not end in `.jar`. But it's still a good change for readability.